realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ProgressListenerTests.java

```diff
@@ -190,7 +190,6 @@ public void onChange(Progress progress) {
         userRealm.close();
         userWithData.logout();
         adminUser.logout();
-        worker.join();
```

Comment:
It concerns me quite a lot that this is a problem. This should only ever be a problem if the progress listeners do not work and the `case 2` is not triggered. It shouldn't be fixed in this PR, but please add a FIXME here instead so we remember to follow up on it.